cmd/helm/repo_add.go

```diff
@@ -83,11 +90,21 @@ func newRepoAddCmd(out io.Writer) *cobra.Command {
 	f.StringVar(&o.keyFile, "key-file", "", "identify HTTPS client using this SSL key file")
 	f.StringVar(&o.caFile, "ca-file", "", "verify certificates of HTTPS-enabled servers using this CA bundle")
 	f.BoolVar(&o.insecureSkipTLSverify, "insecure-skip-tls-verify", false, "skip tls certificate checks for the repository")
+	f.BoolVar(&o.allowDeprecatedRepos, "allow-deprecated-repos", false, "by default, this command will not allow adding official repos that have been permanently deleted. This disables that behavior")
 
 	return cmd
 }
 
 func (o *repoAddOptions) run(out io.Writer) error {
+	// Block deprecated repos
+	if !o.allowDeprecatedRepos {
+		for _, url := range deprecatedRepos {
+			if strings.Contains(o.url, url) {
+				return fmt.Errorf("Repo %q is no longer available", o.url)
```

Comment:
Should we tell them about the new location here?

Reply:
So if you run it, you'll see that the line immediately above it says what the new repo is. That message comes out of the root function as a warning. This one is just the error that stops the repo add.

Reply:
Oh -- you're right... that warning comes up AFTER you add the repo